pyphare/pyphare_tests/test_pharesee/test_geometry.py

```diff
@@ -0,0 +1,528 @@
+
+        if ilvl == 0:
+            lvl_particles = coarse_particles
+        else:
+            level_domain_box = boxm.refine(domain_box, refinement_ratio)
+            lvl_ghost_domain_box = boxm.grow(level_domain_box, domain_layout.particleGhostNbr(interp_order))
+            lvl_particles = Particles(box = lvl_ghost_domain_box)
+
+        if ilvl not in patch_datas:
+            patch_datas[ilvl] = []
+
+        for box in lvl_box:
+
+            ghost_box = boxm.grow(box, 5)
+            origin = box.lower * lvl_cell_width
+            layout = GridLayout(box, origin, lvl_cell_width, interp_order)
+
+            datas = {"Bx": bx(ghost_box, lvl_cell_width, domain_size, origin),
+                     "By": by(ghost_box, lvl_cell_width, domain_size, origin),
+                     "Bz": bz(ghost_box, lvl_cell_width, domain_size, origin),
+                     "Ex": ex(ghost_box, lvl_cell_width, domain_size, origin),
+                     "Ey": ey(ghost_box, lvl_cell_width, domain_size, origin),
+                     "Ez": ez(ghost_box, lvl_cell_width, domain_size, origin),
+                     "particles": lvl_particles.select(ghost_box)
+                     }
+
+            boxed_patch_datas = {}
+            for qty_name, data in datas.items():
+                if qty_name == 'particles':
+                    pdata = ParticleData(layout, data)
+                else:
+                    pdata = FieldData(layout, qty_name, data)
+
+                boxed_patch_datas[qty_name] = pdata
+
+            patch_datas[ilvl].append(boxed_patch_datas)
+
+    patches = {ilvl: [] for ilvl in list(patch_datas.keys())}
+    for ilvl, lvl_patch_datas in patch_datas.items():
+
+
+        for patch_datas in lvl_patch_datas:
+            patches[ilvl].append(Patch(patch_datas))
+
+    patch_levels = {}
+    for ilvl, lvl_patches in patches.items():
+        patch_levels[ilvl] = PatchLevel(ilvl, lvl_patches)
+
+    sorted_levels_numbers = sorted(patch_levels)
+    patch_levels = {ilvl: patch_levels[ilvl] for ilvl in sorted_levels_numbers}
+    return PatchHierarchy(list(patch_levels.values()), domain_box, refinement_ratio)
+
+
+
+
+@ddt
+class GeometryTest(unittest.TestCase):
+
+    def setUp(self):
+
+        nbr_cells = 65
+        origin = 0.
+        interp_order = 1
+        domain_size = 1.
+        cell_width = domain_size / nbr_cells
+        refinement_ratio = 2
+        refinement_boxes = {"L0": {"B0": [(5,), (29,)], "B1": [(32,), (55,)]}}
+
+        self.hierarchy = build_hierarchy(nbr_cells=nbr_cells,
+                               origin=origin,
+                               interp_order=interp_order,
+                               domain_size=domain_size,
+                               cell_width=cell_width,
+                               refinement_ratio=refinement_ratio,
+                               refinement_boxes=refinement_boxes)
+
+
+
+
+    def test_overlaps(self):
```

Comment:
I do not get the test... while I understand it computes several overlaps in various cases to make sure the calculated overlaps are ok

Reply:
the fixture create a hierarchy from this config : 

```
        nbr_cells = 65
        origin = 0.
        interp_order = 1
        domain_size = 1.
        cell_width = domain_size / nbr_cells
        refinement_ratio = 2
        refinement_boxes = {"L0": {"B0": [(5,), (29,)], "B1": [(32,), (55,)]}}
```

then the test calls : 

```
        overlaps = hierarchy_overlaps(self.hierarchy)
```

and check if the calculated overlaps equate the expected